### PR TITLE
Release improvements

### PR DESCRIPTION
**Pull request by mnonnenmacher**
See the commit messages for details.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -54,4 +54,8 @@ jobs:
       env:
         GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
       run: |
-        gh release create $ORT_SERVER_VERSION --notes-file RELEASE_NOTES.md
+        PRERELEASE_ARG=""
```
Comment (sschuberth) [inline .github/workflows/release.yml:57]: Nit: You could even omit this, as interpolating a non-existing variable will also result in the empty string.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -54,4 +54,8 @@ jobs:
...
+        if [[ "$ORT_SERVER_VERSION" == *RC* ]]; then
```
Comment (sschuberth) [inline .github/workflows/release.yml:58]: Should the pattern be `-RC` for safety? Or even `[[ "$ORT_SERVER_VERSION" =~ "-RC[0-9]+" ]]` to align a bit with line 7?

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -48,13 +48,14 @@ jobs:
       run: ./gradlew --stacktrace --no-configuration-cache publishAndReleaseToMavenCentral
 
     - name: Generate Release Notes
-      # Temporary implementation for the very first release. The release notes for this will be written by hand as they
-      # would otherwise contain the complete commit history.
-      run: echo "Initial release!" > RELEASE_NOTES.md
-      #run: ./gradlew -q printChangeLog > RELEASE_NOTES.md
+      run: ./gradlew -q printChangeLog > RELEASE_NOTES.md
```
Comment (mmurto) [inline .github/workflows/release.yml:51]: This seems to print the commits in groups, right? We could consider something more sophisticated like https://keepachangelog.com/en/1.0.0/ as well. I'd say especially as we are quite strict with the commits, using all commits for changelog is not that nice.

Comment (sschuberth) [reply .github/workflows/release.yml:51]: > This seems to print the commits in groups, right?

Yes, it's the same mechanism as at https://github.com/oss-review-toolkit/ort/releases.

Comment (mnonnenmacher) [reply .github/workflows/release.yml:51]: I believe that proper user-facing release notes have to be handwritten, but for now I'm happy to at least have a fully automated solution.

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -54,4 +54,7 @@ jobs:
       env:
         GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
       run: |
-        gh release create $ORT_SERVER_VERSION --notes-file RELEASE_NOTES.md
+        if [[ "$ORT_SERVER_VERSION" =~ "-RC[0-9]+" ]]; then
```
Comment (sschuberth) [inline .github/workflows/release.yml:55]: In my tests, the regex must not be quoted to match, so omit `"` around `-RC[0-9]+`.

**Review by sschuberth — CHANGES_REQUESTED** [review]

**Review by sschuberth — APPROVED** [review]

**Pull request merged**